CONTRIBUTING.md

```diff
@@ -0,0 +1,69 @@
+# Contributing guide
+
+**Want to contribute? Great!** 
+We try to make it easy, and all contributions, even the smaller ones, are more than welcome.
+This includes bug reports, fixes, documentation, examples... 
+But first, read this page.
+
+## Legal
+
+All original contributions to Narayana are licensed under the
+[GNU LESSER GENERAL PUBLIC LICENSE Version 2.1](https://github.com/jbosstm/narayana/blob/master/LICENSE).
+
+All contributions are subject to the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).
+The DCO text is also included verbatim in the [dco.txt](dco.txt) file in the root directory of the repository.
+
+## Reporting an issue
+
+This project uses the [JBTM issue tracker](https://issues.redhat.com/projects/JBTM)
+to manage the issues. Open an issue directly in this issue tracker.
+
+If you believe you found a bug, and it's likely possible, please indicate a way to reproduce it, what you are seeing and what you would expect to see.
+Don't forget to indicate your Narayana, Java and Maven version. 
+
+### Building master
+
+See the [README](README.md) file in the root directory of the repository.
+
+## Before you contribute
+
+To contribute, use GitHub Pull Requests, from your **own** fork.
+
+Also, make sure you have set up your Git authorship correctly:
+
+```
+git config --global user.name "Your Full Name"
+git config --global user.email your.email@example.com
+```
+
+If you use different computers to contribute, please make sure the name is the same on all your computers.
+
+### Code reviews
+
+All submissions, including submissions by project members, need to be reviewed before being merged.
```

Comment:
I think this is "should be reviewed" I have pushed to master before. If we really want to say need to be reviewed then I suppose self review is fine but I expect our release process at least expects to push unreviewed commits up to master